### PR TITLE
Update jammy image

### PR DESCRIPTION
**Pull request by rbradford**
* scripts: Add a script to automate the custom image construction
* docs: Point at custom image build script in documentation
* tests: Update version of Jammy image in use
* tests: Disable test_tpm test
* tests: Give test_vfio_user more RAM

### CODE REVIEW

File: scripts/build-custom-image.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+EOF
+sudo cp script mnt
+sudo chmod +x mnt/script
+sudo chroot mnt ./script
+sudo mv mnt/etc/resolv.conf.backup mnt/etc/resolv.conf
+sudo umount mnt
+sudo kpartx -d $IMAGE_NAME_BASE.raw
+cp $IMAGE_NAME_BASE.raw $IMAGE_NAME_BASE-custom-$(date "+%Y%m%d")-0.raw
+qemu-img convert -p -f raw -O qcow2 $IMAGE_NAME_BASE-custom-$(date "+%Y%m%d")-0.raw $IMAGE_NAME_BASE-custom-$(date "+%Y%m%d")-0.qcow2
+popd
```
Comment (TimePrinciple) [inline scripts/build-custom-image.sh:35]:
```suggestion
popd || exit
```

ditto :)

**Review by TimePrinciple — COMMENTED** [review]
Some suggestions :)

**Comment by jinankjain** [comment]
@smit-gardhariya Please review if anything is missing for kexec test case.

**Comment by TimePrinciple** [comment]
I think `SC2048` should be `SC2046`, a typo :(

My apologies!

File: scripts/build-custom-image.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+mkdir -p custom-image
+pushd custom-image || exit
+wget https://cloud-images.ubuntu.com/jammy/current/jammy-server-cloudimg-amd64.img
+export IMAGE_NAME_BASE=jammy-server-cloudimg-amd64
+qemu-img convert -p -f qcow2 -O raw $IMAGE_NAME_BASE.img $IMAGE_NAME_BASE.raw
+mkdir -p mnt
+export ROOTFS=/dev/mapper/$(sudo kpartx -v -a $IMAGE_NAME_BASE.raw | grep "p1 " | cut -f 3 -d " ")
+sudo mount $ROOTFS mnt
+sudo mv mnt/etc/resolv.conf mnt/etc/resolv.conf.backup
```
Comment (up2wing) [inline scripts/build-custom-image.sh:14]: Maybe adding a blank line here would make the script seems clearer.

File: scripts/build-custom-image.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+echo "nameserver 1.1.1.1" > /etc/resolv.conf
+export DEBIAN_FRONTEND=noninteractive
+apt update
+apt install -y fio iperf iperf3 socat stress cpuid tpm2-tools kexec-tools
+apt remove -y --purge snapd pollinate
+umount /dev/pts
+umount /proc
+history -c
+exit
+EOF
```
Comment (up2wing) [inline scripts/build-custom-image.sh:29]: ditto.

File: scripts/build-custom-image.sh [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+wget https://cloud-images.ubuntu.com/jammy/current/jammy-server-cloudimg-amd64.img
+export IMAGE_NAME_BASE=jammy-server-cloudimg-amd64
+qemu-img convert -p -f qcow2 -O raw $IMAGE_NAME_BASE.img $IMAGE_NAME_BASE.raw
+mkdir -p mnt
+export ROOTFS=/dev/mapper/$(sudo kpartx -v -a $IMAGE_NAME_BASE.raw | grep "p1 " | cut -f 3 -d " ")
+sudo mount $ROOTFS mnt
+sudo mv mnt/etc/resolv.conf mnt/etc/resolv.conf.backup
+cat >script <<EOF
+#!/bin/bash
+set -x
```
Comment (russell-islam) [inline scripts/build-custom-image.sh:17]: why not setting -e here?

**Pull request merged**